### ISSUE TITLE
Exclusion not working for ksp-generated code if added to sourceSets

### ISSUE THREAD

**Issue by jakoss** [jeremymailen/kotlinter-gradle#244]
I have api models annotated with Moshi. I'm using ksp moshi plugin to generate adapters. I would like to add those generated files to Android Studio view, but still skip them in ktlint linting.

The setup of Kotlinter looks like this
```kotlin
tasks.named<org.jmailen.gradle.kotlinter.tasks.LintTask>("lintKotlinMain") {
    exclude {
        it.file.toString().contains("generated")
    }
}
tasks.named<org.jmailen.gradle.kotlinter.tasks.FormatTask>("formatKotlinMain") {
    exclude {
        it.file.toString().contains("generated")
    }
}
```

It works great until i do this:
```kotlin
this.applicationVariants.all {
    val variantName = name
    sourceSets {
        getByName(variantName) {
            java.srcDir(File("build/generated/ksp/$variantName/kotlin"))
        }
    }
}
```

After adding generated sources the `lintKotlin` is complaining about the files under `build/generated`. So it looks like it completely ignoring the `exclude` configuration in this case. Is that Kotlinter error or should i look elsewhere?

**Comment by jeremymailen** [comment]
Please note that the exclusions, when done the way you are doing them, are relative to the _package_ root: https://github.com/jeremymailen/kotlinter-gradle#customizing-tasks

Can you try this suggestion and see if it works: https://github.com/jeremymailen/kotlinter-gradle/issues/242#issuecomment-1074473916

The other alternative is, since Kotlinter creates a lint task per sourceSet, you could define a separate sourceSet for your generated Moshi classes and disable that entire subtask.

**Comment by jakoss** [comment]
Ok, so i did this:

```kotlin
tasks.withType<org.jmailen.gradle.kotlinter.tasks.LintTask> {
    setSource(source - fileTree("$buildDir/generated"))
}
tasks.withType<org.jmailen.gradle.kotlinter.tasks.FormatTask> {
    setSource(source - fileTree("$buildDir/generated"))
}
```

And it solved the issue for all modules except the app one (it's the main android app module). I think that's because of flavors, that are messing up the source sets somehow. So i'm struggling with that. But otherwise - step in good direction

**Comment by jakoss** [comment]
Ok, i think i figured that out. Configuration for android library module (`com.android.build.gradle.LibraryExtension`)
```kotlin
android.libraryVariants.all {
    val variantName = name
    tasks.getByName<org.jmailen.gradle.kotlinter.tasks.LintTask>("lintKotlin${variantName.capitalize()}") {
        setSource(source - fileTree("$buildDir/generated"))
    }
    tasks.getByName<org.jmailen.gradle.kotlinter.tasks.FormatTask>("formatKotlin${variantName.capitalize()}") {
        setSource(source - fileTree("$buildDir/generated"))
    }
    sourceSets {
        getByName(variantName) {
            java.srcDir(File("build/generated/ksp/$variantName/kotlin"))
        }
    }
}
```

and for application module:
```kotlin
android.applicationVariants.all {
    val variantName = name
    tasks.getByName<org.jmailen.gradle.kotlinter.tasks.LintTask>("lintKotlin${variantName.capitalize()}") {
        setSource(source - fileTree("$buildDir/generated"))
    }
    tasks.getByName<org.jmailen.gradle.kotlinter.tasks.FormatTask>("formatKotlin${variantName.capitalize()}") {
        setSource(source - fileTree("$buildDir/generated"))
    }
    sourceSets {
        getByName(variantName) {
            java.srcDir(File("build/generated/ksp/$variantName/kotlin"))
        }
    }
}
```

hope that helps somebody in future :)
